### PR TITLE
Feat: calculate total payment amount to date for ia project funding form

### PR DESCRIPTION
**Pull request by tmastrom**
Card: https://github.com/bcgov/cas-cif/issues/1419

The expenses and payments tracker will be shown in the summary component except when showing a diff of changes. This functionality is described in card [1231](https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1231)

### CODE REVIEW

File: schema/test/unit/computed_columns/form_change_calculated_total_payment_amount_to_date.sql [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  'cif',
+  'project_summary_report',
+  8,
+  4,
+  'project_summary_report',
+  '{"projectSummaryReportPayment":30}'
+);
+
+
+-- TODO: test other operations? eg. if archive?
```
Comment (BCerki) [inline schema/test/unit/computed_columns/form_change_calculated_total_payment_amount_to_date.sql:106]:
```suggestion
```

File: schema/test/unit/computed_columns/form_change_calculated_total_payment_amount_to_date.sql [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+select is(
+  (
+    select cif.form_change_calculated_total_payment_amount_to_date((select row(form_change.*)::cif.form_change from cif.form_change where id=1))
+  ),
+  80.00,
+  'Returns the the calculated total payment amount when only the maximum funding amount is filled in.'
```
Comment (BCerki) [inline schema/test/unit/computed_columns/form_change_calculated_total_payment_amount_to_date.sql:112]: I'd also add tests for:
- when the summary payment is filled in but the max funding isn't
- when summary payment and/or max funding is archived

**Review by BCerki — COMMENTED** [review]
Sorry I didn't notice this in the first round; it looks like the new field isn't showing up on the summary page unless I'm missing something:
image

**Review by BCerki — DISMISSED** [review]

**Review by BCerki — APPROVED** [review]

**Review by BCerki — APPROVED** [review]

File: app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx [hunk 3]
```diff
@@ -204,6 +210,7 @@ describe("The Project Funding Agreement Form Summary", () => {
   it("Only displays the data fields that have changed for an IA form", () => {
```
Comment (BCerki) [inline app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx:210]: In this one and the similar EP one, you could additionally assert that the payments tracker isn't showing

File: app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx [hunk 4]
```diff
@@ -115,24 +119,26 @@ const mockQueryPayloadIA = {
...
+                  maxFundingAmount: 501,
                   provinceSharePercentage: 60,
-                  anticipatedFundingAmount: 300,
+                  anticipatedFundingAmount: 200,
                   proponentCost: 100,
                   contractStartDate: "2021-02-02T23:59:59.999-07:00",
-                  projectAssetsLifeEndDate: "2021-12-31T23:59:59.999-07:00",
+                  projectAssetsLifeEndDate: "2023-03-28T14:41:23.626132-07:00",
                 },
+                calculatedTotalPaymentAmountToDate: "511.0",
```
[the remark this answers is not in the document]
Comment (tmastrom) [reply app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx:129]: I don't have a good reason. I was just poking things

File: app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx (hunk 3 above)
Comment (tmastrom) [reply app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx:210]: good idea

File: app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx [hunk 5]
```diff
@@ -45,6 +45,10 @@ const mockQueryPayloadEP = {
                   contractStartDate: "2021-02-02T23:59:59.999-07:00",
                   projectAssetsLifeEndDate: "2021-12-31T23:59:59.999-07:00",
                 },
+                eligibleExpensesToDate: "1.00",
+                holdbackAmountToDate: "0.00",
+                netPaymentsToDate: "1.00",
+                grossPaymentsToDate: "1.00",
```
[the remark this answers is not in the document]
Comment (tmastrom) [reply app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx:51]: I [created a card](https://github.com/bcgov/cas-cif/issues/1592) to handle this in e2e testing for EP projects. The calculated total payment amount for IA forms is tested in cypress in this PR.

File: app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx (hunk 3 above)
Comment (tmastrom) [reply app/tests/unit/components/Form/ProjectFundingAgreementFormSummary.test.tsx:211]: the payments tracker is no longer a thing

**Review by BCerki — APPROVED** [review]

**Pull request merged**